dotnet/src/Connectors/Connectors.AI.OpenAI/ChatCompletion/AzureChatCompletion.cs

```diff
@@ -31,7 +31,7 @@ public sealed class AzureChatCompletion : AzureOpenAIClientBase, IChatCompletion
         string modelId,
         string endpoint,
         string apiKey,
-        HttpClient? httpClient = null,
+        HttpClient httpClient,
```

Comment:
I don't love having the HttpClient be required... I suspect most developers won't care or understand why they have to provide their own here. (As you say, most should likely use the "WithOpenAI..." extension methods, but some people are using these directly for access to the new streaming mehods).

The other thing that jumps out at me, is that with `httpClient` being required and `organization` being optional, you kind of split the openAI API args around an infrastructure arg. If we stick with required httpClient, then I'd recommend creating overloads with/without the "string organization" arg before httpClient, instead of giving it a default value.

That said, my preference would be to keep httpClient optional if possible; as Stephen says, share an assembly instance of handlers. And just make sure that, where we're calling it with helpers and from KernelBuilder, we're passing it in with the kernel's settings/defaults.

Reply:
Make sense, the changes are rolled back.